lib/modules/userInfo.js

```diff
@@ -142,6 +126,33 @@ addModule('userInfo', function (module, moduleID) { $.extend(true, module, {
 		var utctime = data.created_utc;
 		var d = new Date(utctime * 1000);
 		data.name = data.name.toLowerCase();
+
+		var header = $('<div />');
+		header.append('<a href="/user/' + escapeHTML(data.name) + '">/u/' + escapeHTML(data.name) + '</a> (<a href="/user/' + escapeHTML(data.name) + '/submitted/">Links</a>) (<a href="/user/' + escapeHTML(data.name) + '/comments/">Comments</a>)');
+		
+		// Add friend button to header.
+		RESUtils.getUserInfo(function(me) {
+			var friendButton, myID = 't2_' + me.data.id;
+			// Check cached value first, or fall back to initial server response.
+			if (data.res_cache_friend || (data.is_friend && data.res_cache_friend === undefined)) {
```

Comment:
Oh that will be nice. It felt a bit dirty modifying properties, but it was the only thing that worked for me.